README.md

```diff
@@ -113,9 +113,9 @@ Supported commands are:
     #PRINT s.            (; print the string s. ;)
 
 The supported evaluation strategies are:
- - `SNF` (strong normal form),
+ - `SNF` (strong normal form: a term `t` is in `SNF` if no reduction can occur in `t`),
  - `HNF` (head normal form),
- - `WHNF` (weak head normal form).
+ - `WHNF` (weak head normal form, a term `t` is said in `WHNF` if there is a finite sequence `t=t0`, `t2`, ..., `tn` such that `tn` is in normal form, for all `i`, `ti` reduces to `t(i+1)` and no reduction occurs at the head).
```

Comment:
1) By normal form, do you mean strong normal form?

2) What is the difference between HNF and WHNF according to your definition?

Personally, I don't like this definition because it gives an implementation detail to the user that he should not be aware of. I think the more we are abstract on the strategies, the better it is. This way, we would have more freedom to change the rewrite engine in the future.

Reply:
I don't know anyone but Dedukti's manual calling strong normal form a term which cannot reduce. It is usual to call it normal form, and I find it confusing to use the adjective "strong" since it has no link with "strong normalisation".
Where are the implementation detail, in this definition ? I don't like the notion, but it is not ambiguous and don't depend on the Dedukti strategy.

Reply:
I find it weird to mention a normal form `tn` since it may not exist.
I think a more precise (but even less concise) characterization of `tn` would be:
```for all further reduction sequence starting at `tn`, the reductions never occur at the head.```
or
```all of `tn` reducts are not reducible at the head.```

Essentially: "There exists a way to reduce `t` without touching the head such that all subsequent reductions never touch the head either."

Reply:
I agree with you that this suggestion is bad since it is not clear that `#WHNF t.` computes a reduct of `t` which is in weak-head normal form if it exists, and run forever otherwise.
However, I still claim that my definition of what is a `WHNF` is right (ie. It is concise, capture exactly the notion implemented and is unspecific enough to be independent of the implementation of it in Dedukti). I, of course, might be wrong.
In the case where `t` is not weakly normalizing (ie. `tn` in normal form does not exist) then there is no reduct of `t` which fits with the definition given.